.github/workflows/functional.yml

```diff
@@ -0,0 +1,133 @@
+name: Functional Testing
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
+jobs:
+  functional-tests:
+    name: Gateway Functional Tests
+    runs-on: ubuntu-22.04
+    strategy:
+        matrix:
+          k8s-version: ["1.23.17", "latest"]
+          nginx-image: [nginx, nginx-plus]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Setup Golang Environment
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: stable
+
+      - name: Set GOPATH
+        run: echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+
+      - name: Docker Buildx
+        uses: docker/setup-buildx-action@0d103c3126aa41d772a8362f6aa67afac040f80c # v3.1.0
+        with:
+          driver-opts: network=host
+
+      - name: Output Variables
+        id: vars
+        run: echo "go_path=$(go env GOPATH)" >> $GITHUB_OUTPUT
+
+      - name: Fetch Cached Artifacts
```

Comment:
What are we fetching here? The pipeline shows that nothing is found, and I'm not sure what it's used for.

Reply:
For some reason, i thought we stored the docker images in cache and they needed to be fetched to be used in the later steps. I am testing some things and seeing if we actually don't need this step.

Reply:
i didn't need this step, i have removed it.